src/api/app/services/webui_controller_service/user_checker.rb

```diff
@@ -19,6 +19,7 @@ def find_or_create_user!
       if user.exists?
         User.find_by!(login: user_login)
       else
+        # This will end up in a before_action(on: :create) that updates last_logged_in_at.
```

Comment:
Will it? When?

Reply:
Sorry, there is a typo in the comment. It should have been `before_validation(on: :create)` not `before_action(on: :create)`. And following the trace:  `User.create_user_with_fake_pw!` calls `create!`, that calls `before_validation(on: :create)` which updates `last_logged_in_at` among others.

[create_user_with_fake_pw!](https://github.com/openSUSE/open-build-service/blob/fb6f6202ea5c161607dd24f459a0cb9292b0b6dc/src/api/app/models/user.rb#L170)

[before_validate](https://github.com/openSUSE/open-build-service/blob/fb6f6202ea5c161607dd24f459a0cb9292b0b6dc/src/api/app/models/user.rb#L148)